modules/basicfuncs/list-funcs.c

```diff
@@ -139,6 +139,19 @@ _list_slice(gint argc, GString *argv[], GString *result,
   list_scanner_deinit(&scanner);
 }
 
+static void
+_adjust_list_index(gint argc, GString *argv[],
+                   gint *ndx)
+{
+  gint count = -1;
+
+  if (*ndx < 0)
+    count = _list_count(argc, argv);
+
+  if (*ndx < 0)
```

Comment:
Didn't you mean? `count > 0`

Reply:
This was originally modeled after _adjust_list_slice(), which only triggered the count calculation
if either first_ndx or last_ndx was negative. With this one, only the ndx can be negative, so no point in having both conditionals. I have now merged them.

Thanks for catching this.